### DISCUSSION TITLE
XY panel as a node utilisation heatmap

### DISCUSSION THREAD

**Discussion by rhowe** [grafana/grafana#57213] [Dashboards, Panels & Visualizations]
Maybe other people have ideas for things they'd like to see in the XY panel, so I kept the title vague.

I'd like to plot CPU and memory usage over time for a node - CPU utilisation on one axis, memory usage on another.

This works, but it would be even better if the panel supported specifing the units for each axis.

Also, it would be great to allow colouration of the points based on the age of the timestamp, giving greater prominence to recent values.

An alternative would be to divide the plot area into a grid and colour the grid squares based on number of points within the squares, esentially making a frequency heatmap, but one where the x axis isn't time.

**Comment by leeoniya** [comment]
this should already be possible.

instead of a timestamp you'll probably want to convert that to a numeric "age in hours" column in your data. you can then map `age` to a color or opacity dimension in `Manual` field mapping mode. if you actually need to show the calendar date in the tooltip, you'll want to format it as a string and return it with your data (related: https://github.com/grafana/grafana/issues/57221)

you can control the units, x and y axis labels through normal field overrides.

if you have an example dataset to attach with the columns described above, i can put together a dashboard for you.

**Reply by rhowe** [reply]
Oh, interesting. Field overrides for units weren't working for me, but perhaps that's because we're using 9.1.1. I'll give it another go on our pre-release env which is running 9.2.0.

**Reply by nmarrs** [reply]
Hi @rhowe, were you able to use field overrides successfully?

**Reply by rhowe** [reply]
Yes, thanks. I have a nice XY scatter with CPU on one axis and memory on the other, with appropriate units too. This is using Grafana 9.3.2. I haven't tried the colour overrides yet, but I think it should be doable.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
